Prevent ServiceControllerTests from dispose when already disposed

safern opened the pull request:
cc: @danmosemsft

danmosemsft commented:
ConstructWithServiceName_ToUpper (maybe others) have this same issue eg 

https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/test~2Ffunctional~2Fcli~2F/build/20170828.01/workItem/System.ServiceProcess.ServiceController.Tests/analysis/xunit/System.ServiceProcess.Tests.ServiceControllerTests~2FConstructWithServiceName_ToUpper

I was adding logging to figure out whether it's trying to remove the service twice, but I can't get the tests to run successfully locally. Even in an admin prompt, I get access denied.

safern commented:
We can try adding more logging to the tests and leave them running for a week until they fail again and we get that logging to diagnose them. Also I will try to run them locally.

Added the no merge label because I will scan for all the issues for this test project and disable all the flaky tests inside this project.

danmosemsft commented:
This is the logging I was going to add, but I can't test it since the tests don't work for me yet:

https://github.com/dotnet/corefx/compare/master...danmosemsft:test.servicecontroller?expand=1

safern commented:
I've updated the PR to prevent from disposing the test class when already disposed. Also added the logging you suggested to repro if RemoveService is being called twice.

With this we can let the tests run for some days with this changes and see if this fixes the failures or if at least we get more information of what is happening as I didn't have a  local repro.

danmosemsft commented:
@safern looks like we hit it again, but the logging for calling remove twice didn't trigger. So the failure seems to be because the service was not correctly installed in the first place?

https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/test~2Ffunctional~2Fcli~2F/build/20171005.01/workItem/System.ServiceProcess.ServiceController.Tests/wilogs

safern commented:
There must be a way to know if the service is installed before deleting it right? We should probably do that.  But I wonder, if the service wasn't installed correctly wouldn't it have failed and throw an exception?

chick-fil-a-21 commented:
Why not disable it all until there is a full implementation nt services, which doesn't link to .net 4.7?  One might suggest the real value of .net core is self contained deploys.  Otherwise, just use .Net 4.7

danmosemsft commented:
@chick-fil-a-21 this package System.ServiceProcess.ServiceController already works on .NET Core without a .NET Framework dependency. (It just isn't on Nuget yet) Perhaps I misunderstand your comment?

chick-fil-a-21 commented:
Yes, but I don’t see any actual implementations of nt services in .net core.

From: Dan Moseley <notifications@github.com>
Reply-To: dotnet/corefx <reply@reply.github.com>
Date: Monday, October 23, 2017 at 2:37 PM
To: dotnet/corefx <corefx@noreply.github.com>
Cc: John Davis <john.davis@accesscfa.com>, Mention <mention@noreply.github.com>
Subject: Re: [dotnet/corefx] Prevent ServiceControllerTests from dispose when already disposed (#24042)


@chick-fil-a-21<https://github.com/chick-fil-a-21> this package System.ServiceProcess.ServiceController already works on .NET Core without a .NET Framework dependency. (It just isn't on Nuget yet) Perhaps I misunderstand your comment?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/dotnet/corefx/pull/24042#issuecomment-338755722>, or mute the thread<https://github.com/notifications/unsubscribe-auth/Ad89LVRt7Q6lyrmLeKD--1UAFMhwZpFaks5svNzdgaJpZM4PX86u>.

danmosemsft commented:
@chick-fil-a-21 what service do you expect .NET Core to implement?
